app/services/get-local-mods.ts

```diff
@@ -30,81 +30,214 @@ function getOwml(owmlPath: string) {
+          name: modDirectoryName,
+          uniqueName: modDirectoryName,
+          localVersion: '-',
+          addons: [],
+        });
       }
+    });
+  }
+
+  if (cmowaPath) {
+    localMods.push(getCmowa(cmowaPath));
+  }
+
+  if (alphaPath) {
+    const manifestAlphaPaths = globby.sync(`**/manifest.json`, {
+      cwd: `${alphaPath}/BepInEx/plugins`,
+      absolute: true,
+    });
 
+    localMods.push(getBepInEx(alphaPath));
+
+    manifestAlphaPaths.forEach((manifestPath) => {
+      const modPath = path.dirname(manifestPath);
       try {
-        mod.isEnabled = isEnabled(mod);
+        const { manifest, missingAttributes } = manifestPartialToFull(
+          fs.readJsonSync(manifestPath)
+        );
+
+        const modWithSameId = localMods.find(
+          (localMod) => localMod.uniqueName === manifest.uniqueName
+        );
+        if (modWithSameId) {
+          modWithSameId.errors.push(
+            modsText.duplicateModError(manifest.uniqueName)
+          );
+          return;
+        }
+
+        const mod: Mod = {
+          name: manifest.name,
+          author: manifest.author,
+          uniqueName: manifest.uniqueName,
+          localVersion: coerce(manifest.version)?.version ?? manifest.version,
+          modPath,
+          errors: [],
+          dependencies: manifest.dependencies ?? [],
+          warning: manifest.warning,
+          patcher: manifest.patcher,
+          conflicts: manifest.conflicts,
+          pathsToPreserve: manifest.pathsToPreserve,
+          addons: [],
+          isAlpha: true,
+        };
+
+        if (missingAttributes.length > 0) {
+          mod.errors.push(
+            modsText.missingManifestAttributesError(
+              manifestPath,
+              missingAttributes
+            )
+          );
+        }
+
+        try {
+          mod.isEnabled = isEnabled(mod);
+        } catch (error) {
+          mod.isEnabled = true;
+          debugConsole.error(error);
+        }
+
+        localMods.push(mod);
       } catch (error) {
-        mod.isEnabled = true;
-        debugConsole.error(error);
+        const modDirectoryName = path.basename(modPath);
+        localMods.push({
+          author: modDirectoryName,
+          dependencies: [],
+          errors: [modsText.brokenManifestError(modDirectoryName, `${error}`)],
+          modPath,
+          name: modDirectoryName,
+          uniqueName: modDirectoryName,
+          localVersion: '-',
+          addons: [],
+          isAlpha: true,
+        });
       }
-
-      localMods.push(mod);
-    } catch (error) {
-      const modDirectoryName = path.basename(modPath);
-      localMods.push({
-        author: modDirectoryName,
-        dependencies: [],
-        errors: [modsText.brokenManifestError(modDirectoryName, `${error}`)],
-        modPath,
-        name: modDirectoryName,
-        uniqueName: modDirectoryName,
-        localVersion: '-',
-        addons: [],
-      });
-    }
-  });
+    });
+  }
```

Comment:
Everything inside the `if (alphaPath)` check is just a copy-paste from the `if (owmlPath)` block, with very minor differences. Extract it to a separate function to avoid repeating so much code. And try to do the same for the rest of your changes if possible.